### ISSUE TITLE
[CT-699] [Bug] Source freshness enabled config doesn't read variables

### ISSUE THREAD

**Issue by fivetran-joemarkiewicz** [dbt-labs/dbt-core#5311]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When leveraging the `sources` config to disable sources within a package, I am noticing that the use of variables with the intent to be overwritten does not work. See the example below:

In my root `dbt_project.yml`
![image](https://user-images.githubusercontent.com/74217849/171257512-b809186c-3551-4de1-ad8f-0ee27ec9a3f5.png)

### Expected Behavior

The default variables is required otherwise I get the following error.
```zsh
18:20:35  Encountered an error:
Compilation Error
  Could not render {{ var('jira_using_versions') }}: Required var 'jira_using_versions' not found in config:
  Vars supplied to <Configuration> = {}
```

Therefore, I set the default value to true, but found that this default value is the only value that is used. I would expect dbt would use the variable I declare in the `vars` section. This may not be that big of a deal of individual dbt projects. However, this is pretty crucial for dbt packages to leverage various variables and allow users to override them. It seems users cannot override these source variables however.

### Steps To Reproduce

- Override a variable in your dbt_project.yml
- Use that variable to disable a source via the `sources` config.
  - You will need to set a default value, but use a different value than the one you set in the `vars` config. 
- Run `dbt source freshness` and see the variable you declared is not used.

### Relevant log output

```shell
See above.
```

### Environment

```markdown
- OS: MacOS M1
- Python: 3.8.9
- dbt: 1.1.0
```

### What database are you using dbt with?

bigquery

### Additional Context

Similar to what I said above. You could just hard code the values for the sources and see it works fine. However, for packages we leverage the variables so users can change the behavior of the package. I would expect using the same variable to disable features in other parts of the package would also be able to disable the source itself.

**Comment by gshank** [comment]
Are you trying to override a source that is provided by a different package (i.e. not your root project)? Because in order for source configs to override a source from a different package, you need to set the attribute: "override: <package_name>". https://docs.getdbt.com/reference/resource-properties/overrides

**Comment by jtcohen6** [comment]
@fivetran-joemarkiewicz The issue here is that `vars` defined in `dbt_project.yml` aren't available for setting configs that are also defined in `dbt_project.yml`, since the former and the latter are parsed at the same time. This has definitely come up before: https://github.com/dbt-labs/dbt-core/issues/4873, https://github.com/dbt-labs/dbt-core/issues/4938

If you're actually defining the sources in the package, you can use those variables in the sources' `.yml` definitions directly:
```yml
# models/src_jira.yml
version: 2
sources:
  - name: jira 
    tables:
      - name: sprint
        description: Table of all sprints.
        config:
          enabled: "{{ var('jira_using_sprints', true) }}"
```

If you're trying (from a separate root project) to disable a single source table from an installed package, your better bet is to use a simple boolean `false`.

**Comment by fivetran-joemarkiewicz** [comment]
@jtcohen6 this is exactly what I was looking for! I had attempted the below and noticed it wasn't working. It looks like I just needed the `config`. Thanks for sharing!
```yml
# models/src_jira.yml
version: 2
sources:
  - name: jira 
    tables:
      - name: sprint
        description: Table of all sprints.
        enabled: "{{ var('jira_using_sprints', true) }}"
```

Is this behavior only available in dbt-core >= v1.1.0?

Also, thank you for sharing the #4938 discussion! That is incredibly interesting and I look forward to seeing the vibe evolution of vars 😄

**Comment by jtcohen6** [comment]
@fivetran-joemarkiewicz Glad to hear that's working! And correct, this is v1.1.0+ only
